lib/elixir/lib/task.ex

```diff
@@ -768,35 +786,44 @@ defmodule Task do
         end
 
         def init(_opts) do
-          # We will keep all running tasks in a map
-          {:ok, %{tasks: %{}}}
+          # We will keep the refs and tags for each task in maps.
+          {:ok, %{task_refs: %{}, task_tags: %{}}}
         end
 
         # Imagine we invoke a task from the GenServer to access a URL...
         def handle_call(:some_message, _from, state) do
-          url = ...
+          url = "..."
           task = Task.Supervisor.async_nolink(MyApp.TaskSupervisor, fn -> fetch_url(url) end)
 
-          # After we start the task, we store its reference and the url it is fetching
-          state = put_in(state.tasks[task.ref], url)
+          # After we start the task, we store its reference and the URL it is fetching
+          state = put_in(state.task_refs[task.ref], url)
+          # We also store its tag and map it to the reference
+          state = put_in(state.task_tags[task.tag], task.ref)
```

Comment:
```suggestion
          state = put_in(state.task_refs[task.ref], {url, task.tag})
          # We also store its tag and map it to the reference
          state = put_in(state.task_tags[task.tag], {url, task.ref})
```

Or we store the url in only one of them but it is important for tag to point to ref and vice-versa.